src/Wizard/WizardSteps/WizardStep/WizardStep.tsx

```diff
@@ -34,42 +36,16 @@ export default function WizardStep({
   );
 }
 
-// prop types
-WizardStep.propTypes = {
-  /**
-   * If true, the step is marked as completed. Is normally set by the WizardSteps parent component.
-   */
-  completed: PropTypes.bool,
-  /**
-   * The error text to display.
-   * If this prop is set, the helperText prop is ignored.
-   */
-  errorText: PropTypes.node,
-  /**
-   * The helper text to display.
-   * If the errorText prop is set, this prop is ignored.
-   */
-  helperText: PropTypes.node,
-  /**
-   * The index of the step. Normally set by the WizardSteps parent component.
-   */
-  index: PropTypes.number,
-  /**
-   * The label displayed in the step icon.
-   */
-  label: PropTypes.string.isRequired,
-  /**
-   * If true, the step is marked as last. Normally set by the WizardSteps parent component.
-   */
-  last: PropTypes.bool
-};
-
 /**
  * The WizardStepLabel is a helper component for the WizardStep component. It provides functionality for the helperText and errorText props. It is a separate component so that it can be used in the MUI Step component and hence access the context state.
  */
-function WizardStepLabel({ label, helperText, errorText }) {
+function WizardStepLabel({
+  label,
+  helperText,
+  errorText
+}: Pick<WizardStepProps, "label" | "helperText" | "errorText">) {
   // get context state
-  const { active, completed } = useStepContext();
+  const { active, completed } = useStepContext() as StepContextType;
```

Comment:
Why do we need to use as here? Isn't this MUI hook already typed?

Reply:
It is typed, but it is typed as `StepContextType | {}`. The documented reason for this in code is that technically if you use the `useStepContext` component not within its context provider, then you'll get an empty object. Makes sense. 

The problem is I have found no way to type-guard against an empty object, so I can't get typescript to acknowledge that I can access the `active` and `completed` props. MUI gets around this by the fact that their code is written in js, and the type are written alongside. 

Given that WizardStepLabel is always rendered within a `Step`, and a `Step` always [generates the context](https://github.com/mui/material-ui/blob/1b9c8ab5a7c0125fa411f46ffafba76f04fc6eae/packages/mui-material/src/Step/Step.js#L111-L120) I gave up sinking any more time into this.

[Typescript playground](https://www.typescriptlang.org/play?strictNullChecks=false&noImplicitReturns=false#code/C4TwDgpgBAysFgMIHsB28AewoF4oG8AoKEqAQwGNgBLANwgC4oAjZZAGwjNQG5jSKyALZhO8ACZNWHLr0IBfQoUGoAztmRgaaAIK4ClGvSbAATgFcIAGiiCRYiJKgAzMu1UR55VQS8AfWHgkNExgPkIAegioc1VqVABzKDR2EChgAAtodniIH2YIdmQAdxtVYQgMkqghMgBraFBIVQpTai0Y1HEIU3VucXSMsmwAeS1qXShqH2RndPBoOAQUdAgsG36oVGQNOabofEVCIA) if you care to have a play.

Reply:
https://www.typescriptlang.org/play?strictNullChecks=false&noImplicitReturns=false#code/C4TwDgpgBAysFgMIHsB28AewoF4oG8AoKEqAQwGNgBLANwgC4oAjZZAGwjNQG5jSKyALZhO8ACZNWHLr0IBfQoUGoAztmRgaaAIK4ClGvSbAATgFcIAGiiCRYiJKgAzMu1UR55VQS8AfWHgkNExgPkIAegioc1VqVABzKDR2EChgAAtodniIH2YIdmQAdxtVYQgMkqghMgBraFBIVQpTai0Y1HEIU3VucXSMsmwAeS1qXShqH2RndPBoOAQUdAgsG36oVGQNOabofEVqOYAKAHI7UQgJM6nUZPHdAEoCfhIorer9lraOuu3ijNHqg9EJYtgCoFliE1sA3rY0OoEfZro4AJLAACyw2geE02hBADpLg5xHx5EA